tools/castle-editor/code/framedesign.pas

```diff
@@ -1857,6 +1863,29 @@ procedure TDesignFrame.ControlsTreeDragOver(Sender, Source: TObject; X,
   ControlsTree.Invalidate; // force custom-drawn look redraw
 end;
 
+procedure TDesignFrame.ControlsTreeEditing(Sender: TObject; Node: TTreeNode;
+  var AllowEdit: Boolean);
+begin
+  { this event is fired when calling TCustomListView.CanEdit which itself is called in TCustomListView.ShowEditor
+    therefore this event preceeds initializing and showing of the editor }
+  //here we have to "restore" the pure name of the component (without class name) before starting edit
+  Node.Text := TComponent(ControlsTree.Selected.Data).name;
+end;
```

Comment:
Should this maybe use `TComponent(Node.Data)` instead `TComponent(ControlsTree.Selected.Data)`?

It seems more reliable to use the argument `Node` as the thing that indicates the component, instead of relaying on `ControlsTree.Selected` (and assuming that this event is always called for selected node). Note that the tree hierarchy also supports selecting multiple components with Ctrl.

Minor: `Name` instead of `name`.

Reply:
I've changed to `Node.Data`, and it seems to work without any issues, so indeed this is more correct way of handling this. Also tested for renaming by pressing F2 and clicking - both work as expected and correctly.
Done in 92d304b2b6020559f9e91ebc16b3b473860afbd9